### PR TITLE
fix(TDP-5558): no loading feedback when overwriting a dataset

### PR DESCRIPTION
**Pull request by ncomont**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-5558

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

File: dataprep-webapp/src/app/services/state/confirm/confirm-state-service.spec.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  https://github.com/Talend/data-prep/blob/master/LICENSE
+
+  You should have received a copy of the agreement
+  along with this program; if not, write to Talend SA
+  9 rue Pages 92150 Suresnes, France
+
+  ============================================================================*/
+
+describe('confirm state service', () => {
+	let stateMock;
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/state/confirm/confirm-state-service.spec.js]: ?

File: dataprep-webapp/src/app/services/state/confirm/confirm-state-service.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+		confirmState.texts = texts;
+	}
+
+	hide() {
+		confirmState.visible = false;
+		confirmState.texts = [];
+	}
+
+	reset() {
+		confirmState.visible = false;
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/state/confirm/confirm-state-service.js]: just call hide instead

File: dataprep-webapp/src/app/services/import/import-service.spec.js [hunk 3]
```diff
@@ -617,12 +617,12 @@ describe('Import service', () => {
...
 				// then
 				expect(DatasetService.checkNameAvailability).toHaveBeenCalledWith(ImportService.datasetName);
-				expect(TalendConfirmService.confirm).toHaveBeenCalledWith(['UPDATE_EXISTING_DATASET'], { dataset: 'my cool dataset' });
+				expect(ConfirmService.confirm).toHaveBeenCalledWith(['UPDATE_EXISTING_DATASET'], { dataset: 'my cool dataset' });
 				expect(DatasetService.create).not.toHaveBeenCalled();
 				expect(DatasetService.update).not.toHaveBeenCalled();
 			}));
 
-			it('should create dataset with modified name', inject(($rootScope, $q, TalendConfirmService, DatasetService, ImportService) => {
+			it('should create dataset with modified name', inject(($rootScope, $q, ConfirmService, DatasetService, ImportService) => {
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/import/import-service.spec.js]: useless import I guess

File: dataprep-webapp/src/app/services/confirm/confirm-service.spec.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+			a: 'translation_a',
+			b: 'translation_b',
+			c: 'translation_c',
+		});
+		$translateProvider.preferredLanguage('en');
+	}));
+
+	beforeEach(inject((StateService) => {
+		spyOn(StateService, 'showConfirmationModal').and.returnValue();
+		spyOn(StateService, 'hideConfirmationModal').and.returnValue();
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/confirm/confirm-service.spec.js]: useless?

File: dataprep-webapp/src/app/services/confirm/confirm-service.spec.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+		$translateProvider.translations('en', {
+			a: 'translation_a',
+			b: 'translation_b',
+			c: 'translation_c',
+		});
+		$translateProvider.preferredLanguage('en');
+	}));
+
+	beforeEach(inject((StateService) => {
+		spyOn(StateService, 'showConfirmationModal').and.returnValue();
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/confirm/confirm-service.spec.js]: You don't have to return value

File: dataprep-webapp/src/app/services/confirm/confirm-service.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+	 * @returns {promise} Promise that resolves (validate) or reject (refuse/cancel) the choice
+	 * @description Show the confirm modal element and return a promise that will be resolve on button click or modal dismiss
+	 * Example : ConfirmService.confirm(['First text', 'Second text'], {translateArg: 'value'})
+	 */
+	confirm(textIds, textArgs) {
+		this.$translate(textIds, textArgs)
+			.then(translations => textIds.map(id => translations[id]))
+			.then(this.StateService.showConfirmationModal);
+
+		return new Promise((res, rej) => {
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/confirm/confirm-service.js]: Did you test on IE 11?

File: dataprep-webapp/src/app/services/confirm/confirm-service.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+ * @description Confirm service. This service provide the entry point to the confirmation modals.
+ */
+export default class ConfirmService {
+	constructor(state, StateService, $translate) {
+		'ngInject';
+
+		this.state = state;
+		this.StateService = StateService;
+		this.$translate = $translate;
+		this.resolve = null;
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/confirm/confirm-service.js]: Where is it used?

Comment (ncomont) [reply dataprep-webapp/src/app/services/confirm/confirm-service.js]: In the template

File: dataprep-webapp/src/app/services/state/confirm/confirm-state-service.js (hunk 2 above)
Comment (ncomont) [reply dataprep-webapp/src/app/services/state/confirm/confirm-state-service.js]: The goal of this method is to reset the state, not to hide the modal. Even if in this case the behaviour is the same, I think that these two method should not have any relation.

File: dataprep-webapp/src/app/services/state/confirm/confirm-state-service.spec.js (hunk 1 above)
Comment (ncomont) [reply dataprep-webapp/src/app/services/state/confirm/confirm-state-service.spec.js]: Fixed, thx

File: dataprep-webapp/src/app/services/import/import-service.spec.js (hunk 3 above)
Comment (ncomont) [reply dataprep-webapp/src/app/services/import/import-service.spec.js]: ✅

File: dataprep-webapp/src/app/services/confirm/confirm-service.spec.js (hunk 4 above)
Comment (ncomont) [reply dataprep-webapp/src/app/services/confirm/confirm-service.spec.js]: ✅

File: dataprep-webapp/src/app/services/confirm/confirm-service.spec.js (hunk 5 above)
Comment (ncomont) [reply dataprep-webapp/src/app/services/confirm/confirm-service.spec.js]: ✅

File: dataprep-webapp/src/app/services/confirm/confirm-service.js (hunk 6 above)
Comment (ncomont) [reply dataprep-webapp/src/app/services/confirm/confirm-service.js]: Yep

**Pull request merged**